pandas/tests/plotting/test_frame.py

```diff
@@ -2650,11 +2650,20 @@ def test_pie_df(self):
             self._check_colors(ax.patches, facecolors=color_args)
 
     def test_pie_df_nan(self):
+        import matplotlib as mpl
+
         df = DataFrame(np.random.rand(4, 4))
         for i in range(4):
             df.iloc[i, i] = np.nan
         fig, axes = self.plt.subplots(ncols=4)
-        df.plot.pie(subplots=True, ax=axes, legend=True)
+
+        # GH 37668
+        if mpl.__version__ >= "3.3":
+            kwargs = {"normalize": True}
+        else:
+            kwargs = {}
+
+        df.plot.pie(subplots=True, ax=axes, legend=True, **kwargs)
```

Comment:
since we are explicitly fixing the warning, should we do something like to ensure the warning is gone?
```python
with tm.assert_produces_warning(None):
    df.plot.pie(subplots=True, ax=axes, legend=True, **kwargs)
```

Reply:
Thank you for the hint!
Since, the warning is relevant only for the newer matplotlib versions, thus I implemented checking that no warning is raised only for this case.
Meanwhile for the older version there is no need to check for warnings.
There is a slight code repetition, which I can easily overcome (make kwargs dict), if required.
How does it look like?

Reply:
I slightly prefer to use kwargs, and we should check no warning is produced regardless of mpl version, so something like below is enough for me (instead of only doing the warning check on mpl>3.3):
```python
kwargs = {}
if mpl.__version__ >= "3.3":
     kwargs = {"normalize": True}

with tm.assert_produces_warning(None):
    df.plot.pie(subplots=True, ax=axes, legend=True, **kwargs)
```

how does it look to you?